pkg/analysis/passes/osvscanner/osvscanner_test.go

```diff
@@ -0,0 +1,254 @@
+		Report: interceptor.ReportInterceptor(),
+	}
+	_, err := Analyzer.Run(pass)
+	require.NoError(t, err)
+	require.Len(t, interceptor.Diagnostics, 0)
+}
+
+// TestEmptyResultsReportAll
+func TestEmptyResultsReportAll(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+	pass := &analysis.Pass{
+		RootDir: filepath.Join("./"),
+		ResultOf: map[*analysis.Analyzer]interface{}{
+			archive.Analyzer:    filepath.Join("testdata", "golang", "none"),
+			sourcecode.Analyzer: filepath.Join("testdata", "golang", "none"),
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+	// Turn on ReportAll for all rules, then turn it back off at the end of the test
+	reportAll(Analyzer)
+	t.Cleanup(func() {
+		undoReportAll(Analyzer)
+	})
+	_, err := Analyzer.Run(pass)
+	require.NoError(t, err)
+	require.Len(t, interceptor.Diagnostics, 2)
+	require.Equal(t, "Binary for osv-scanner was found in PATH", interceptor.Diagnostics[0].Title)
+	require.Equal(t, "osv-scanner successfully ran", interceptor.Diagnostics[1].Title)
+}
+
+// TestNoIssueResults
+func TestNoIssueResults(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+	pass := &analysis.Pass{
+		RootDir: filepath.Join("./"),
+		ResultOf: map[*analysis.Analyzer]interface{}{
+			archive.Analyzer:    filepath.Join("testdata", "node", "none"),
+			sourcecode.Analyzer: filepath.Join("testdata", "node", "none"),
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+	_, err := Analyzer.Run(pass)
+	require.NoError(t, err)
+	require.Len(t, interceptor.Diagnostics, 0)
+}
+
+// TestNoIssueResultsReportAll
+func TestNoIssueResultsReportAll(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+	pass := &analysis.Pass{
+		RootDir: filepath.Join("./"),
+		ResultOf: map[*analysis.Analyzer]interface{}{
+			archive.Analyzer:    filepath.Join("testdata", "node", "none"),
+			sourcecode.Analyzer: filepath.Join("testdata", "node", "none"),
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+	// Turn on ReportAll for all rules, then turn it back off at the end of the test
+	reportAll(Analyzer)
+	t.Cleanup(func() {
+		undoReportAll(Analyzer)
+	})
+	_, err := Analyzer.Run(pass)
+	require.NoError(t, err)
+
+	require.Len(t, interceptor.Diagnostics, 3)
+	require.Equal(t, "osv-scanner passed", interceptor.Diagnostics[2].Title)
+}
+
+// TestCriticalSeverityResults
+func TestCriticalSeverityResults(t *testing.T) {
+	var interceptor testpassinterceptor.TestPassInterceptor
+	pass := &analysis.Pass{
+		RootDir: filepath.Join("./"),
+		ResultOf: map[*analysis.Analyzer]interface{}{
+			archive.Analyzer:    filepath.Join("testdata", "node", "critical"),
+			sourcecode.Analyzer: filepath.Join("testdata", "node", "critical"),
+		},
+		Report: interceptor.ReportInterceptor(),
+	}
+	_, err := Analyzer.Run(pass)
+	require.NoError(t, err)
+	require.Len(t, interceptor.Diagnostics, 2)
+	require.Equal(t, "osv-scanner detected a critical severity issue", interceptor.Diagnostics[0].Title)
```

Comment:
I had troubles with this approach (using diagnostics's indexes directly) in the past because these scanners results are not always in the same order.

Check this https://github.com/grafana/plugin-validator/blob/main/pkg/analysis/passes/jssourcemap/jssourcemap_test.go#L47

Reply:
very nice, i'll update the test cases to use that method. might need to have the len() to have "Greater than or Equal to" since the vulnerabilities can increase over time.

Reply:
ended up using the same type of check, but with subset instead, or i'd have to enumerate all of the output.